web/src/main/java/org/cbioportal/web/ReferenceGenomeGeneController.java

```diff
@@ -0,0 +1,86 @@
+package org.cbioportal.web;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.cbioportal.model.ReferenceGenomeGene;
+import org.cbioportal.service.ReferenceGenomeGeneService;
+import org.cbioportal.service.exception.GeneNotFoundException;
+import org.cbioportal.web.config.annotation.PublicApi;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.constraints.Size;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@PublicApi
+@RestController
+@Validated
+@Api(tags = "Reference Genome Genes", description = " ")
+public class ReferenceGenomeGeneController {
+
+    private static final int GENE_MAX_PAGE_SIZE = 100000;
+    private static final String GENE_DEFAULT_PAGE_SIZE = "100000";
+    
+    @Autowired
+    private ReferenceGenomeGeneService referenceGenomeGeneService;
+
+    @RequestMapping(value = "/reference-genome-genes/{genomeName}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Get all reference genes")
+    public ResponseEntity<List<ReferenceGenomeGene>> getAllRefGenes(
+        @ApiParam(required = true, value = "Name of Reference Genome hg19")
+        @PathVariable String genomeName)  {
+        
+        return new ResponseEntity<>(
+                referenceGenomeGeneService.fetchAllReferenceGenomeGenes(genomeName), HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/reference-genome-genes/{genomeName}/{geneId}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Get a gene of a reference genome of interest")
+    public ResponseEntity<ReferenceGenomeGene> getGene(
+        @ApiParam(required = true, value = "Name of Reference Genome hg19")
+        @PathVariable String genomeName,
+        @ApiParam(required = true, value = "Entrez Gene ID 207")
+        @PathVariable Integer geneId) throws GeneNotFoundException {
+    
+        return new ResponseEntity<>(referenceGenomeGeneService.getReferenceGenomeGene(geneId, genomeName), HttpStatus.OK);
+    }
+    
+    @RequestMapping(value = "/reference-genome-genes/{genomeName}/fetch", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE,
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Fetch genes of reference genome of interest")
+    public ResponseEntity<List<ReferenceGenomeGene>> fetchGenes(
```

Comment:
Can we rename this method to `fetchReferenceGenomeGenes`? We have another `fetchGenes` method in `GeneController` and when you auto generate API client, you get function names like `fetchGenesUsingPOST_1` due to name conflicts.

When implementing controllers, try to match the name of your functions to the name of your model.

Reply:
@onursumer, I see. change made, thanks for the tip.